x/posts/types/models/post.go

```diff
@@ -106,13 +108,33 @@ type Post struct {
 	OptionalData   OptionalData   `json:"optional_data,omitempty" yaml:"optional_data,omitempty"` // Arbitrary data that can be used from the developers
 	Creator        sdk.AccAddress `json:"creator" yaml:"creator"`                                 // Creator of the Post
 	Attachments    Attachments    `json:"attachments,omitempty" yaml:"attachments,omitempty"`     // Contains all the attachments that are shared with the post
-	PollData       *PollData      `json:"poll_data,omitempty" yaml:"poll_data"`                   // Contains the poll details, if existing
+	PollData       *PollData      `json:"poll_data,omitempty" yaml:"poll_data,omitempty"`         // Contains the poll details, if existing
 }
 
-func NewPost(id, parentID PostID, message string, allowsComments bool, subspace string,
+// computeID computes a post ID based on the content of the given post.
+// nolint
+func computeID(parentID PostID, message, subspace string, allowsComments bool, creationTime time.Time, creator sdk.AccAddress,
+	attachments Attachments, pollData *PollData) PostID {
```

Comment:
What about passing a `Post` to `computeID` instead of all the separated data? I think this might be easier to maintain in the future if other data must be added. What do you think?

Reply:
I was thinking about that too, I'll try